docs/en/stack/ml/nlp/ml-nlp-text-emb-vector-search-example.asciidoc

```diff
@@ -0,0 +1,339 @@
+[[ml-nlp-text-emb-vector-search-example]]
+= How to deploy a text embedding model and use it with vector search
+
+++++
+<titleabbrev>Text embedding and vector search</titleabbrev>
+++++
+:keywords: {ml-init}, {stack}, {nlp}
+
+You can use these instructions to deploy a 
+<<ml-nlp-text-embedding,text embedding>> model in {es}, test the model, and 
+add it to an {infer} ingest pipeline. It enables you to generate vector 
+representations of text and perform vector similarity search on the generated 
+vectors. The model that is used in the example is publicly available on 
+https://huggingface.co/[HuggingFace].
+
+The example uses a public dataset from the 
+https://microsoft.github.io/msmarco/#ranking[MS MARCO Passage Ranking Task]. It 
+consists of real questions from the Microsoft Bing search engine and human 
+generated answers for them. The example works with a sample of this dataset, 
```

Comment:
This is two words per https://brand.elastic.co/302f66895/p/194a3b-writing-style-guide
```suggestion
generated answers for them. The example works with a sample of this data set, 
```